Source/Utils/Average.H

```diff
@@ -57,38 +52,146 @@ namespace Average{
+        int num_pts[3];
+        // Index of first point of source Array4 from which interpolation is done
+        int idx_min[3];
+
+        // Compute number of points to loop over (either 1 or 2)
+        // and starting indices of fine array in each direction
+        for ( int l = 0; l < 3; ++l ) {
+            if ( cr[l] == 1 ) {
+                num_pts[l] = 1+abs(sf[l]-sc[l]);
+                idx_min[l] = ic[l]-sc[l]*(1-sf[l]);
+            }
+            else {
+                num_pts[l] = 2-sf[l];
+                idx_min[l] = ic[l]*cr[l]+static_cast<int>(cr[l]/2)*(1-sc[l])-(1-sf[l]);
+            }
+        }
+
+        // Auxiliary integer variables
+        const int numx = num_pts[0];
+        const int numy = num_pts[1];
+        const int numz = num_pts[2];
+        const int imin = idx_min[0];
+        const int jmin = idx_min[1];
+        const int kmin = idx_min[2];
+
+        // Interpolate over points computed above
+        Real c = 0.0_rt;
+        for (int kref = 0; kref < numz; ++kref) {
+            for (int jref = 0; jref < numy; ++jref) {
+                for (int iref = 0; iref < numx; ++iref) {
+                    c += mf_fp_arr(imin+iref,jmin+jref,kmin+kref,comp);
+                }
+            }
+        }
+        return c / (numx*numy*numz);
+    };
+
+    /**
+     * \brief Loops over the boxes of the coarsened MultiFab \c mf_cp and fills
+     *        them by interpolating the data contained in the fine MultiFab \c mf_fp.
+     *
+     * \param[in,out] mf_cp      coarsened MultiFab containing the floating point data
+     *                           to be filled by interpolating the input fine MultiFab
+     * \param[in]     mf_fp      fine MultiFab containing the floating point data to be interpolated
+     * \param[in]     dcomp      offset for the fourth component of the coarsened Array4
+     *                           object, extracted from its MultiFab, where the interpolated
+     *                           values will be stored
+     * \param[in]     scomp      offset for the fourth component of the fine Array4
+     *                           object, extracted from its MultiFab, containing the
+     *                           data to be interpolated
+     * \param[in]     ncomp      number of components to loop over for the coarsened
+     *                           Array4 extracted from the coarsened MultiFab \c mf_cp
+     * \param[in]     crse_ratio coarsening ratio between the fine MultiFab \c mf_fp
+     *                           and the coarsened MultiFab \c mf_cp
+     */
+    void CoarsenAndInterpolateLoop ( MultiFab& mf_cp,
+                                     const MultiFab& mf_fp,
+                                     const int dcomp,
+                                     const int scomp,
+                                     const int ncomp,
+                                     const IntVect crse_ratio );
```

Comment:
Can you provide a default a value for IntVect crse_ratio as amrex::IntVect(AMREX_D_DECL(1,1,1)). For cases, where, interpolation is needed without coarsening.

Reply:
Sure, I will do that. `amrex::IntVect(1)` should be enough for this purpose (see `amrex/Src/Base/AMReX_IntVect.H:100`).

Reply:
Oh! that is neater! 👍